src/test/java/org/opendevstack/provision/services/BitbucketAdapterTest.java

```diff
@@ -302,22 +322,26 @@ public void callCreateRepoApiTest() throws Exception {
 
     Mockito.doReturn(basePath).when(spyAdapter).getAdapterApiUri();
 
-    Mockito.doReturn(expected).when(client).callHttp(anyString(), any(), anyBoolean(),
-        eq(RestClient.HTTP_VERB.POST), any());
+    //    Mockito.doReturn(expected)
+    //        .when(restClient)
+    //        .callHttp(anyString(), any(), anyBoolean(), eq(RestClient.HTTP_VERB.POST), any());
+
+    mockExecute(matchesClientCall().method(HttpMethod.POST)).thenReturn(expected);
 
     Mockito.doNothing().when(spyAdapter).setRepositoryPermissions(any(), any(), any(), any());
 
     RepositoryData actual = spyAdapter.callCreateRepoApi(projectKey, repo);
 
-    Mockito.verify(spyAdapter).setRepositoryPermissions(eq(expected), eq(projectKey), eq("groups"),
-        any());
+    Mockito.verify(spyAdapter)
+        .setRepositoryPermissions(eq(expected), eq(projectKey), eq("groups"), any());
 
-    Mockito.verify(spyAdapter).setRepositoryPermissions(eq(expected), eq(projectKey), eq("users"),
-        any());
+    Mockito.verify(spyAdapter)
+        .setRepositoryPermissions(eq(expected), eq(projectKey), eq("users"), any());
 
-    Mockito.verify(client).callHttp(eq(uri), eq(repo), anyBoolean(), eq(RestClient.HTTP_VERB.POST),
-        any());
+    //    Mockito.verify(restClient)
```

Comment:
And the rest client question. ;-)

Reply:
Ok, please see my last commits on this PR.